collector/cpu_linux.go

```diff
@@ -104,11 +104,20 @@ func (c *cpuCollector) updateCPUfreq(ch chan<- prometheus.Metric) error {
 
 	var value uint64
 
+	cpu_core_throttles := make(map[int]uint64)
+
 	// cpu loop
 	for _, cpu := range cpus {
 		_, cpuName := filepath.Split(cpu)
 		cpuNum := strings.TrimPrefix(cpuName, "cpu")
 
+		core_id := -1
+		if value, err := readUintFromFile(filepath.Join(cpu, "topology/core_id")); err != nil {
```

Comment:
Since we only need `core_id` for thermal throttles, let's move this read to after we check for thermal throttles.  Then we can avoid the `-1` handling.

Reply:
I moved it to an outer if as there is no need to read thermal throttles if we don't have the core_id.

CI is running....